dynamic_array.py

```
@@ -1,41 +1,40 @@
 # Hacker Rank Problem Link: https://www.hackerrank.com/challenges/dynamic-array/problem
 
-#!/bin/python3
+def seqIndex(x , lastAns , n):
+    return (x ^ lastAns) % n
 
-import os
-import sys
 
-#
-# Complete the dynamicArray function below.
-#
 def dynamicArray(n, queries):
-    #
-    # Write your code here.
-    #
+   
     lastAns = 0
     seqList = [[]]* n
+    value = 0
+    
     for query in queries:
         if query[0] == 1:
-            seqIndex  = (query[1] ^ lastAns) % n
-            seqList[seqIndex] = seqList[seqIndex] + [query[2]]
+            seqList[seqIndex(query[1], lastAns , n)] = seqList[seqIndex(query[1], lastAns , n)] + [query[2]]
         elif query[0] == 2:
-            seqIndex = (query[1] ^ lastAns) % n
-            value = query[2] % len(seqList[seqIndex])
-            lastAns = seqList[seqIndex][value]
+            try:
+                value = query[2] % len(seqList[seqIndex(query[1], lastAns , n)])
+            except ZeroDivisionError:
+                 print ("You can't divide by zero")
+                 break
+            lastAns = seqList[seqIndex(query[1], lastAns , n)][value]
             print(lastAns)
+        else:
+            return None

```

That should raise an exception as it is not an expected input.